pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -335,6 +338,14 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
 			instance.Endpoint.EndpointPort = uint32(port)
 
 			localCluster := cb.buildInboundClusterForPortOrUDS(instance, endpointAddress)
+			localCluster.UpstreamBindConfig = &core.BindConfig{
```

Comment:
To avoid infinite loop due to iptables

Reply:
It should be required. 
Otherwise when envoy establishes connection to <podip>:8080, 
for this inbound flow you don't want to capture, 
for other cases outbound traffic which endpoint in EDS happened to be the same <podip>:8080,
iptables needs extra attributes to recognize.

Bind magical local ip was the low hanging fruit